### DISCUSSION TITLE
getDiagnostics for AssetStorage

### DISCUSSION THREAD

**Discussion by Quillraven** [libktx/ktx#343] [Ideas]
I am wondering if a getDiagnostics method like the one from AssetManager is possible for AssetStorage.

My current use case is that some assets are loaded/unloaded throughout the game and it is not happening in only one place like a LoadingScreen.

I was searching for an easy way to debug the AssetStorage when the game is closing to diagnose the ref count of the different assets and to see what is even part of the AssetStorage. 

Is there already a way to do that? If not, is it possible to add?

**Comment by czyzby** [comment]
It's not available right now, but it's possible to implement. I can see how that would be useful.

I'm not a huge fan of methods that return strings for debugging, but I can implement a method returning an `AssetStorageSnapshot` (or something like that) with all the necessary information and with a `toString` method that outputs something similar to `AssetManager#getDiagnostics`.

**Reply by czyzby** [reply]
Anyway, you can use classic debugging right now and check the state of `assets` map within the storage.

EDIT: I've added `AssetStorage#takeSnapshot` and `AssetStorage#takeSnapshotAsync`, they should be available starting from `1.9.14-SNAPSHOT`/`1.9.14-b1`.

**Reply by czyzby** [reply]
See #344.
